src/main/java/com/igot/cios/plugins/upgrad/UpgradPluginServiceImpl.java

```diff
@@ -79,39 +109,148 @@ private UpgradContentEntity saveOrUpdateUpgradContent(String externalId, JsonNod
+                IndexRequest indexRequest = new IndexRequest(Constants.CIOS_CONTENT_INDEX_NAME)
+                        .id(uniqueId)
+                        .source(entityMap);
+
+                bulkRequest.add(indexRequest);
+                log.info("Preparing to add document for externalId: {}", contentEntity.getExternalId());
+            } catch (Exception e) {
+                log.error("Error while processing contentEntity with externalId: {}", contentEntity.getExternalId(), e);
+            }
+        });
+        try {
+            client.bulk(bulkRequest, RequestOptions.DEFAULT);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        Long totalCourseCount = repository.count();
+        JsonNode response = dataTransformUtility.fetchPartnerInfoUsingApi(partnerCode);
+        JsonNode resultData = response.path(Constants.RESULT);
+        JsonNode data = resultData.path(Constants.DATA);
+        ((ObjectNode) data).put(Constants.TOTAL_COURSES_COUNT, totalCourseCount);
+        dataTransformUtility.updatingPartnerInfo(resultData);
     }
 
-    @Override
-    public List<?> fetchAllContent() {
-        return upgradContentRepository.findAll();
+    private void flattenContentData(Map<String, Object> entityMap) {
+        if (entityMap.containsKey("ciosData") && entityMap.get("ciosData") instanceof Map) {
+            Map<String, Object> ciosDataMap = (Map<String, Object>) entityMap.get("ciosData");
+            if (ciosDataMap.containsKey("content") && ciosDataMap.get("content") instanceof Map) {
+                Map<String, Object> contentMap = (Map<String, Object>) ciosDataMap.get("content");
+                entityMap.putAll(contentMap);
+                entityMap.remove("ciosData");
+            }
+        }
     }
 
     @Override
-    public void deleteContent(Object contentEntity) {
-        upgradContentRepository.delete((UpgradContentEntity) contentEntity);
+    public Page<?> fetchAllContentFromSecondaryDb(RequestDto dto) {
+        Pageable pageable= PageRequest.of(dto.getPage(), dto.getSize());
+        return repository.findAllCiosDataAndIsActive(dto.getIsActive(),pageable,dto.getKeyword());
     }
 
     @Override
     public Object readContentByExternalId(String externalid) {
-        Optional<UpgradContentEntity> entity = upgradContentRepository.findByExternalId(externalid);
+        Optional<UpgradContentEntity> entity = repository.findByExternalId(externalid);
         if(entity.isPresent()){
             return entity.get().getCiosData();
         }else{
             throw new CiosContentException("No data found for given id",externalid, HttpStatus.BAD_REQUEST);
         }
+
     }
 
     @Override
     public Object updateContent(JsonNode jsonNode,String partnerCode) {
+        Timestamp currentTime = new Timestamp(System.currentTimeMillis());
+        String externalId=jsonNode.path("content").get("externalId").asText();
```

Comment:
Please use constants.Content and all other places where used string like content, externalId